src/main/java/com/cloudhopper/smpp/impl/DefaultSmppServer.java

```diff
@@ -360,8 +361,17 @@ protected void createSession(Long sessionId, Channel channel, SmppSessionConfigu
         // session created, now pass it upstream
         counters.incrementSessionCreatedAndGet();
         incrementSessionSizeCounters(session);
-        this.serverHandler.sessionCreated(sessionId, session, preparedBindResponse);
-        
+
+        try {
+            this.serverHandler.sessionCreated(sessionId, session, preparedBindResponse);
+        } catch (SessionCreatedFailedException e) {
+            logger.error("sessionCreated call on serverHandler failed. Unable to finish the binding. " +
```

Comment:
maybe it will be useful to log full classname of `serverHandler`, so we can see what a exact handler failed to process `sessionCreated`